### ISSUE TITLE
c.enclosingPosition microaggressions

### ISSUE THREAD

**Issue by acdenhartog** [scalamacros/paradise#76]
Possibly the most frivolous issue ever:

Could the `c.enclosingPosition` parameter please be a default for methods that use it? It is so annoying. Especially while the autocomplete still has a ton of deprecated `enclosingWhatever` clogging up the name space. There is even scaladoc comments in `scala.reflect.macros.FrontEnds` to manually explain that `c.enclosingPosition` is the default thing to use.

The following should work without breaking any compatibility:
`def echo(pos: Position = enclosingPosition, msg: String): Unit`

`enclosingPosition` is within scope since `self: blackbox.Context =>` 

Then one can call it as `c.echo(msg="ho ho ho Merry Christmas")`

And it will probably just become `e.echo("and a Happy New Year!")` when the compiler get smarter about named parameters.

**Comment by xeno-by** [comment]
Do you think introducing a new overload with just one parameter would also work for you?

**Comment by acdenhartog** [comment]
Sure, anything will work, but it is more a matter principle. A default parameter means less duplication and shorter files, which makes the API easier read and understand and it makes some of that scaladoc redundant. I would just go for writing msg="..." now and hope for Scalac fix the rest of it later.

**Comment by acdenhartog** [comment]
I noticed only recently that almost every tree has a position and line numbers embedded in it; that is a pretty neat feature, and should perhaps be mentioned in the scaladoc of those echo methods. I also found that passing null as a position works just fine, and can be nicer for debugging too.

In the end I would have preferred the default parameter as `null` for echo and `enclosingPosition` for the others. I would not do overloads because that just further obscures these nice features.

In any case the most important point to me is not less typing but rather getting an API that is more self-documenting so that an IDE can understand and help along.

I would not have bothered you with any of this, but making macros easier to use for everyone was on the roadmap, so...
